serverlessrepo/publish.py

```diff
@@ -36,6 +37,14 @@ def publish_application(template, sar_client=None):
         response = sar_client.create_application(**request)
         application_id = response['ApplicationId']
         actions = [CREATE_APPLICATION]
+
+        if app_metadata.semantic_version:
```

Comment:
After a second thought, I still feel like this wait for latest version to sync does not belong here in publish application method. What if the client does not want to put policy or does not care about whether the latest version is updated at all, then this wait is not necessary. 

I can think of two options:
1. Move the wait into put policy.
1. Make it a separate method.

What do you think?

Reply:
I prefer to make it a separate method, since the wait is only needed when creating an app with version, not really specific to put policy.